### ISSUE TITLE
[BUG] Error: useNavigate() may be used only in the context of a <Router> component.

### ISSUE THREAD

**Issue by doron-cohen** [frontegg/frontegg-react#640]
**Describe the bug**
I get the following error:
```
Error: useNavigate() may be used only in the context of a <Router> component.
```
whether I am wrapping the `FronteggProvider` with `BrowserRouter` or not.

**To Reproduce**

`index.tsx`:
```
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { FronteggProvider } from '@frontegg/react';
import App from './App';

const contextOptions = {
  baseUrl: 'https://***.frontegg.com'
};

ReactDOM.render(
  <React.StrictMode>
    <BrowserRouter>
      <FronteggProvider contextOptions={contextOptions}>
        <App />
      </FronteggProvider>
    </BrowserRouter>
  </React.StrictMode>,
  document.getElementById('root')
);
```

`App.tsx`:
```
import { useAuth } from '@frontegg/react';

function App() {
  const { user, isAuthenticated } = useAuth();

  return (
    <div className="App">
      {isAuthenticated && (
        <div>
          <img src={user.profilePictureUrl} alt={user.name} />

          <span>{user.name}</span>
        </div>
      )}
    </div>
  );
}

export default App;
```

`package.json`:
```
{
  "name": "application",
  "version": "0.0.0",
  "scripts": {
    "dev": "vite",
    "build": "tsc && vite build",
    "preview": "vite preview"
  },
  "dependencies": {
    "@frontegg/react": "^4.0.2",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-router-dom": "^6.2.1"
  },
  "devDependencies": {
    "@types/react": "^17.0.33",
    "@types/react-dom": "^17.0.10",
    "@vitejs/plugin-react": "^1.0.7",
    "typescript": "^4.4.4",
    "vite": "^2.7.2"
  }
}
```

**Additional context**
I am using Vite. I don't think it is related but could be.

**Comment by doron-cohen** [comment]
I just tried version `4.0.3-alpha.1597926679` and it solved it but I get this error in the console:
```
Error: metadata not found: saml
    getMetadata index.ts:8
    fulfilled chunk-XYAVBKPK.js:3227
    promise callback*step chunk-XYAVBKPK.js:3240
    node_modules chunk-XYAVBKPK.js:3242
    __awaiter5 chunk-XYAVBKPK.js:3224
    getMetadata chunk-XYAVBKPK.js:3199
    node_modules chunk-XYAVBKPK.js:3249
    node_modules chunk-XYAVBKPK.js:3242
    __awaiter5 chunk-XYAVBKPK.js:3224
    getSamlMetadata index.ts:20
    Redux 21
    node_modules chunk-TNPGN2ER.js:1660
    fk FronteggApp-efdad027.js:512
    node_modules chunk-XYAVBKPK.js:18373
    gg FronteggApp-efdad027.js:372
    Oj FronteggApp-efdad027.js:511
    ek FronteggApp-efdad027.js:511
    V4 FronteggApp-efdad027.js:243
    node_modules chunk-XYAVBKPK.js:18202
    node_modules chunk-XYAVBKPK.js:18197
    node_modules chunk-XYAVBKPK.js:18414
    __init chunk-JSLGV3TO.js:23
    node_modules @frontegg_react.js:868
    __init chunk-JSLGV3TO.js:23
    node_modules @frontegg_react.js:4748
    __require chunk-JSLGV3TO.js:26
    <anonymous> @frontegg_react:1
react_devtools_backend.js:4045:25
    overrideMethod react_devtools_backend.js:4045
    Redux 26
    StoreListener index-82d481ed.js:1559
    fk FronteggApp-efdad027.js:512
    unstable_runWithPriority FronteggApp-efdad027.js:245
    gg FronteggApp-efdad027.js:372
    Oj FronteggApp-efdad027.js:511
    ek FronteggApp-efdad027.js:511
    V4 FronteggApp-efdad027.js:243
    onmessage FronteggApp-efdad027.js:239
    (Async: EventHandlerNonNull)
    js FronteggApp-efdad027.js:239
    js FronteggApp-efdad027.js:248
    __init chunk-JSLGV3TO.js:23
    js index.js:1
    __init chunk-JSLGV3TO.js:23
    js @frontegg_react.js:4748
    __require chunk-JSLGV3TO.js:26
    <anonymous> @frontegg_react:1
```

**Comment by frontegg-david** [comment]
Hi @doron-cohen, 
Fixed version 4.0.3, we added support for v5 and v6 react router library.

The saml error is not related to useNavigate issue, it's just 404 error for SAML configuration not found and it will display email/password login screens instead of email input with preLogin.


Thanks,
David

**Comment by KaynRonan** [comment]
> Hi @doron-cohen, Fixed version 4.0.3, we added support for v5 and v6 react router library.
> 
> The saml error is not related to useNavigate issue, it's just 404 error for SAML configuration not found and it will display email/password login screens instead of email input with preLogin.
> 
> Thanks, David

so, where is the demo for react-router v6 ? In your code repository, there is a demo, but this demo did not even import the routes.tsx file. https://github.com/frontegg/frontegg-react/blob/master/packages/demo-saas/src/Routes.tsx @frontegg-david
